modules/dnn/test/test_misc.cpp

```diff
@@ -63,6 +63,72 @@ TEST(imagesFromBlob, Regression)
     }
 }
 
+TEST(blobFromImageWithParams_4ch, NHWC_scalar_scale)
+{
+    Mat ch[4];
+    for(int i = 0; i < 4; i++)
+        ch[i] = Mat::ones(10, 10, CV_8U)*i;
+
+    std::vector<double> factorVec = {0.1, 0.2, 0.3, 0.4};
+
+    Scalar scalefactor(factorVec[0], factorVec[1], factorVec[2], factorVec[3]);
+    Mat img;
+    merge(ch, 4, img);
+
+    Image2BlobParams param;
+    param.scalefactor = scalefactor;
+    param.datalayout = DNN_LAYOUT_NHWC;
+    Mat blob = dnn::blobFromImageWithParams(img, param); // [1, 10, 10, 4]
+
+    float* blobPtr = blob.ptr<float>(0);
+
+    std::vector<float> targetVec = {(float )factorVec[0] * 0, (float )factorVec[1] * 1, (float )factorVec[2] * 2, (float )factorVec[3] * 3}; // Target Value.
+    for (int hi = 0; hi < 10; hi++)
+    {
+        for (int wi = 0; wi < 10; wi++)
+        {
+            float* hwPtr = blobPtr + hi * 10 * 4 + wi * 4;
+
+            // Check equal
+            EXPECT_NEAR(hwPtr[0], targetVec[0], 1e-5);
+            EXPECT_NEAR(hwPtr[1], targetVec[1], 1e-5);
+            EXPECT_NEAR(hwPtr[2], targetVec[2], 1e-5);
+            EXPECT_NEAR(hwPtr[3], targetVec[3], 1e-5);
+        }
+    }
```

Comment:
The same idea may be applied here too:
```
cv::Mat expected_target(10, 10, CV_32FC4, cv::Scalar(...));
EXPECT_MAT_NEAR(expected_target, blob, 1e-5);
```

Reply:
I just found that the shape of `cv::Mat expected_target(10, 10, CV_32FC4, cv::Scalar(...));` is different with the output of `blobFromImage`. The main difference is the former is a multi-channel image, while the latter is a multi-dimensional image (channels are moved to dimensions.)